TasksApp/TasksApp/PriorityTask.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+
+namespace TasksApp
+{
+    internal class PriorityTask : WeeklyTask
+    {
+        internal DateTime Date { get; set; }
+        internal TasksPriority Priority { get; set; }
+
+        internal PriorityTask(string name, DateTime data, TasksPriority priority) : base(name)
+        {
+            Date = data;
+            Priority = priority;
+        }
+
+        internal override string GetAlarm()
+        {
+            if (Date.ToShortDateString().Equals(DateTime.MinValue.ToShortDateString(), StringComparison.Ordinal))
```

Comment:
do not compare string compare dates:
```suggestion
            if (Date == default)
```